libethcore/Farm.h

```diff
@@ -122,6 +123,18 @@ class Farm: public FarmFace
 		m_isMining = false;
 	}
 	
+	/**
+	 * @brief Stop all mining activities and Starts them again
+	 */
+	void restart()
+	{
+		stop();
+		start(m_lastSealer, b_lastMixed);
+		
+		assert(m_onMinerRestart);
```

Comment:
Instead of requiring this handler, let's do `if (m_onMinerRestart)`. Then you will not have to set this to empty lambda in other places.